third-party/cmake/DownloadAndExtractFile.cmake

```diff
@@ -74,11 +74,26 @@ list(GET status 0 status_code)
 list(GET status 1 status_string)
 
 if(NOT status_code EQUAL 0)
-  message(FATAL_ERROR "error: downloading '${URL}' failed
+  # Retry on certain errors, e.g. CURLE_COULDNT_RESOLVE_HOST, which is often
+  # seen with libtermkey (www.leonerd.org.uk).
+  if(status_code EQUAL 6)  # "Couldn't resolve host name"
+    message(STATUS "warning: retrying '${URL}' (${status_string}, status ${status_code})")
+    execute_process(COMMAND ${CMAKE_COMMAND} -E sleep 10)
+    file(DOWNLOAD ${URL} ${file}
+      ${timeout_args}
+      ${hash_args}
+      STATUS status
+      LOG log)
+    list(GET status 0 status_code)
+    list(GET status 1 status_string)
+  endif()
+  if(NOT status_code EQUAL 0)
+    message(FATAL_ERROR "error: downloading '${URL}' failed
   status_code: ${status_code}
   status_string: ${status_string}
   log: ${log}
```

Comment:
could put this on 1 line as well

Reply:
> could put this on 1 line as well

I think it is good to parse visually as-is, especially since the log is multiline:
```
  error: downloading
  'http://www.leonerd.org.uk/code/libtermkey/libtermkey-0.21.1.tar.gz' failed
    status_code: 6
    status_string: "Couldn't resolve host name"
    log: Curl_ipv4_resolve_r failed for www.leonerd.org.uk
  Couldn't resolve host 'www.leonerd.org.uk'
  Closing connection 0
```